src/test/java/redis/clients/jedis/tests/JedisPubSubTest.java

```diff
@@ -0,0 +1,87 @@
+package redis.clients.jedis.tests;
+
+
+import static org.junit.Assert.assertEquals;
+
+
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.JedisPubSub;
+import redis.clients.jedis.tests.commands.JedisCommandTestBase;
+import redis.clients.jedis.util.SafeEncoder;
+
+import java.util.List;
+
+
+import static redis.clients.jedis.Protocol.Command.CLIENT;
+public class JedisPubSubTest  {
+
+    @Test
+    public void PubSub()
+    {
+
+        JedisPoolConfig config = new JedisPoolConfig();
+
+        JedisPool pool = new JedisPool(config,"localhost",6379);
+
+        Jedis JedisSub = pool.getResource(); // Jedis Instance which subscribes the channel
+
+        Jedis JedisData = pool.getResource(); // Jedis instance which used for tracking
+
+        String ClientID = String.valueOf(JedisSub.clientId());
+
+        //Runnable for tracking the Client and other functionalities
+        Runnable RunnableTracker = () -> {
+            JedisData.sendCommand(CLIENT, SafeEncoder.encode("TRACKING"),SafeEncoder.encode("on"),
+                    SafeEncoder.encode("REDIRECT"),SafeEncoder.encode(ClientID));
+            JedisData.set("foo","bar");
+            String value = JedisData.get("foo");
+            assertEquals("bar",value);
```

Comment:
This won't work. 

1. If assert throws an exception in a different thread, the main thread won't get that exception and the test would pass
2. This should be outside the thread. With some order of execution, this can return `newFoo`